cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/local/LocalStorage.java

```diff
@@ -0,0 +1,70 @@
+package com.linkedin.openhouse.cluster.storage.local;
+
+import com.linkedin.openhouse.cluster.storage.Storage;
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * The LocalStorage class is an implementation of the Storage interface for local storage. It uses a
+ * LocalStorageClient to interact with the local file system.
+ */
+@Component
+public class LocalStorage implements Storage {
+
+  private static final StorageType.Type TYPE = StorageType.LOCAL;
+
+  @Autowired private StorageProperties storageProperties;
+
+  // Lazy initialization of the LocalStorageClient
+  @Autowired @Lazy private LocalStorageClient localStorageClient;
+
+  /**
+   * Check if the local storage is active.
+   *
+   * <p>The local storage is considered active if the default type is not set or no types are
+   * provided or specific "local" type is provided.
+   *
+   * @return true if the local storage is active, false otherwise
+   */
+  @Override
+  public boolean isActive() {
+    if (storageProperties.getDefaultType() == null) {
```

Comment:
If a default type is not set but storageType is explicitly configured to be a type that is different from local, for example hdfs, in that case storage type should be set to hdfs and not local, right?

Default type should be used only if explicit type is not configured.

Reply:
yes correct! default type will only be configured if explicit type is not configured. I've added more validations+tests for this.

Reply:
probably didn't explain clearly.

If the defaultType == null and type == "hdfs" then storageType will be hdfs and not local, correct?
In that case retuning that local storage is active if storageType == null is incorrect.

Reply:
if `defaultType==null` and `types.hdfs != null` then storageType will not be configured. we should see a boot time exception. I have it lined up in the next PR: https://github.com/HotSushi/openhouse/pull/1/files#diff-3eab08eaf8a381dc4108e6db8d27ff2f34a126442a904758bb0f5ca9211c54bdR34

Reply:
Looks like you are assuming an invariant that if type is configured then default should also be set? Is that right?

May be list down the invariants on the config somewhere, enforce them as you said and assert them here?